src/sparseml/pytorch/optim/modifier_pruning.py

```diff
@@ -678,8 +740,17 @@ def _check_mask_update(self, module: Module, epoch: float, steps_per_epoch: int)
             self._module_masks.enabled = True
             started = True
 
-        if self.end_pending(epoch, steps_per_epoch) and not self._leave_enabled:
-            self._module_masks.enabled = False
+        if self.end_pending(epoch, steps_per_epoch):
+
+            if self._score_type == PruningScoreTypes.MOVEMENT:
```

Comment:
Same as above, this func should be overridden by the mvp modifier.

Reply:
Let's push comments like these off to a refactor later. Priority is to get this landed ASAP for release.

Reply:
this was refactored in the woodfisher PR to call into a single callback from the pruning mask object